### ISSUE TITLE
Keto example permission model doesn't traverse graph

### ISSUE THREAD

**Issue by IchordeDionysos** [ory/docs#1744]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/docs/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/docs/blob/master/CONTRIBUTING.md).
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [X] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

Following steps in this guide: https://www.ory.sh/docs/keto/modeling/create-permission-model

Please let me know if I understand something fundamentally wrong about the way the permission model is supposed to work :)

### Reproducing the bug

1. Go the permission model page
2. Copy code of [`permissions-v5.ts`](https://github.com/ory/docs/blob/a3619d4e961c0f222475301f1faf88b537275cc9/docs/keto/modeling/create-permission-model.mdx?plain=1#L315-L361)
3. Go to https://console.ory.sh/projects/-/permissions/configuration
4. Paste the code under "Permission Rules"
5. Click save
6. Go to relationships: https://console.ory.sh/projects/-/permissions/relationships
7. Create the following two relationships:
  1. `userA` is `owners` of `Folder:folderA`
  2. `folderA` is `parents` of `Document:docA`
8. Make the following request: https://-.projects.oryapis.com/relation-tuples/check?namespace=Document&object=docA&relation=share&subject_id=userA&max-depth=100

Expected: `/relation-tuples/check` returns `{"allowed": true}`
Actual: `/relation-tuples/check` returns `{"allowed": false}`

### Version

v0.13.0-alpha.0

### On which operating system are you observing this issue?

Ory Network

### In which environment are you deploying?

Ory Network

### Additional Context

Permission v3 and v4 don't work. Permission v5 works again (which is marked as optional) :D

**Comment by zepatrik** [comment]
I agree that the docs are confusing and incomplete here. The console is not super helpful either (and partially lacking functionality), but here are the details on how to make it work:

You are mixing subject sets (typed subjects) and subject IDs (untyped). What you want to do is create the relationships with the subject set, which you can do in the console like this:

<img width="919" alt="Screenshot 2024-06-03 at 12 23 25" src="https://github.com/ory/docs/assets/5354445/718db82e-b4cc-47bf-aa2e-739e76511320">

The permission check then also has to use subject sets: `/relation-tuples/check?namespace=Document&object=docA&relation=share&subject_set.object=userA&max-depth=100&subject_set.namespace=User&subject_set.relation=`

In general you should probably never use subject IDs, but always the subject set instead. We do plan to deprecate the subject ID for this confusion and no real benefit.

I will keep this issue open so that we keep track of the docs changes.
